### PR TITLE
Improve Kafka environment variable config documentation

### PR DESCRIPTION
**Pull request by StFS**
Add documenation about dynamic config keys and the need to disambiguate them in the config file or as system properties. Also explain the need to use double underscore for properties that include a period in their key.

### CODE REVIEW

File: site/documentation/content/admin-guide/hono-kafka-client-configuration.md [hunk 1]
```diff
@@ -111,6 +111,22 @@ The Kafka clients require at least the `bootstrap.servers` property to be set. T
...
+variable names. For example, the property keys `a.b`, `a-b` and `a_b` would all map to the environment variable `A_B`.
+To resolve this ambiguity, the configuration framework relies on the correct property being defined in a configuration
+source with a lower presedence than the environment variable configuration source. It's therefore necessary to provide
+any such configuration property in the `application.yml` file or as a Java system property to help the configuration
+system disambiguate the property key, even if only providing a placeholder value. Once that has been done, it's 
+possible to override the value provided with an environment variable.
+{{% /notice %}}
+
+{{% notice tip %}}
+Any of the Kafka configuration options that use a period (`.`) in the key must use a double underscore in the
```
Comment (sophokles73) [inline site/documentation/content/admin-guide/hono-kafka-client-configuration.md:125]: I was under the impression that the double underscores are required to represent the quotes in a property name, or am I mistaken?

Comment (radcortez) [reply site/documentation/content/admin-guide/hono-kafka-client-configuration.md:125]: Correct. If the key does not require quotes, then you don't need the double underscore.

**Review by sophokles73 — COMMENTED** [review]
awesome :+1:

**Review by radcortez — APPROVED** [review]

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**